src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -245,51 +246,64 @@ def _get_resource(client, resource_group_name, *subresources):
             raise
 
 
-def _get_vnet_network_profile(cmd, location, resource_group_name, vnet_name, vnet_address_prefix, subnet, subnet_address_prefix):
+def _get_vnet_network_profile(cmd, location, resource_group_name, vnet, vnet_address_prefix, subnet, subnet_address_prefix):
     from azure.cli.core.profiles import ResourceType
     from msrestazure.tools import parse_resource_id, is_valid_resource_id
 
     containerInstanceDelegationServiceName = "Microsoft.ContainerInstance/containerGroups"
     Delegation = cmd.get_models('Delegation', resource_type=ResourceType.MGMT_NETWORK)
     aci_delegation = Delegation(
-        name="Microsoft.ContainerInstance.containerGroups",
-        service_name="Microsoft.ContainerInstance/containerGroups"
+        name=containerInstanceDelegationServiceName,
+        service_name=containerInstanceDelegationServiceName
     )
 
     ncf = cf_network(cmd.cli_ctx)
 
+    vnet_name = vnet
     subnet_name = subnet
     if is_valid_resource_id(subnet):
         parsed_subnet_id = parse_resource_id(subnet)
         subnet_name = parsed_subnet_id['resource_name']
         vnet_name = parsed_subnet_id['name']
+        resource_group_name = parsed_subnet_id['resource_group']
+    elif is_valid_resource_id(vnet):
+        parsed_vnet_id = parse_resource_id(vnet)
+        vnet_name = parsed_vnet_id['resource_name']
+        resource_group_name = parsed_vnet_id['resource_group']
+        ncf.virtual_networks.get(resource_group_name, vnet_name)
 
     default_network_profile_name = "aci-network-profile-{}-{}".format(vnet_name, subnet_name)
 
     subnet = _get_resource(ncf.subnets, resource_group_name, vnet_name, subnet_name)
     # For an existing subnet, validate and add delegation if needed
     if subnet:
-        for endpoint in (subnet.service_endpoints or []):
-            if endpoint.service != "Microsoft.ContainerInstance":
-                raise CLIError("Can not use subnet with existing service links other than 'Microsoft.ContainerInstance'.")
+        logger.info('Using existing subnet "%s" in resource group "%s"', subnet.name, resource_group_name)
+        for sal in (subnet.service_association_links or []):
+            if sal.linked_resource_type != containerInstanceDelegationServiceName:
+                raise CLIError("Can not use subnet with existing service association links other than {}.".format(containerInstanceDelegationServiceName))
```

Comment:
Why doesn't the service provide this error message?

Reply:
Currently, the error message doesn't propagate back correctly between our side and networking side. This is to make the error clear to customers for now.